chia/cmds/chia.py

```diff
@@ -39,26 +41,63 @@ def monkey_patch_click() -> None:
     context_settings=CONTEXT_SETTINGS,
 )
 @click.option("--root-path", default=DEFAULT_ROOT_PATH, help="Config file root", type=click.Path(), show_default=True)
+@click.option(
+    "--keys-root-path", default=DEFAULT_KEYS_ROOT_PATH, help="Keyring file root", type=click.Path(), show_default=True
+)
+@click.option("--passphrase-file", type=click.File("r"), help="File or descriptor to read the keyring passphase from")
 @click.pass_context
-def cli(ctx: click.Context, root_path: str) -> None:
+def cli(ctx: click.Context, root_path: str, **kwargs) -> None:
     from pathlib import Path
 
     ctx.ensure_object(dict)
     ctx.obj["root_path"] = Path(root_path)
 
+    # keys_root_path and passphrase_file are grabbed from kwargs because they may not be
+    # present. If supports_keyring_passphrase() returns False, we remove those options
+    # from the CLI.
+    keys_root_path = kwargs.get("keys_root_path")
+    if keys_root_path:
+        set_keys_root_path(Path(keys_root_path))
+
+    passphrase_file = kwargs.get("passphrase_file")
+    if passphrase_file:
+        from .passphrase_funcs import cache_passphrase, read_passphrase_from_file
+
+        try:
+            cache_passphrase(read_passphrase_from_file(passphrase_file))
+        except Exception as e:
+            print(f"Failed to read passphrase: {e}")
+
+
+if not supports_keyring_passphrase():
+    from chia.cmds.passphrase_funcs import remove_passphrase_options_from_cmd
+
+    # TODO: Remove once keyring passphrase management is rolled out to all platforms
+    remove_passphrase_options_from_cmd(cli)
```

Comment:
I agree that we will likely want to front-load the experience of migrating users to the new keyring. We still have work to do on the GUI to make that experience smooth (probably some additional CLI work as well).

For the moment, remove_passphrase_options_from_cmd is used to scrub the options from appearing on macOS and Windows (to avoid confusion), since the feature is Linux-only (for now).

Since the new keyring support is disabled by default in this PR, I'd like to revisit the migration topic when the GUI is implemented and we're ready to enable the feature. Does that sound reasonable?

Reply:
yeah that makes sense. 
What is the rationale for merging it in a disabled state? Wouldn't it be simpler to merge it and enable it by default immediately, once the GUI stuff is done?

Reply:
I'd like to checkpoint the work at this point since it's been challenging keeping the branch up to date for so long. The GUI work is being pursued by Lipa, and I believe there's still more to be ironed out there.